src/client.rs

```diff
@@ -95,7 +95,7 @@ pub async fn client_entrypoint(
     mut stream: TcpStream,
     client_server_map: ClientServerMap,
     shutdown: Receiver<()>,
-    drain: Sender<i8>,
+    drain: Sender<i32>,
```

Comment:
Ah yeah, I forgot we're keeping this one alive until the client disconnects. Good catch. I'd make it a `usize` to avoid this problem forever.

Reply:
On second thought....we're only sending `1` and `0` through this channel. Why would it overflow?

Reply:
Ah yes, because `total_clients` becomes a `i8`. Wow! Subtle.